### DISCUSSION TITLE
JavaScript Help

### DISCUSSION THREAD

**Discussion by twilson9034** [Budibase/budibase#3749] [Help] [answered]
I've got a table that has columns for each tire pressures on an asset.  I would like to get a count for all tires that are below 100 for a certain column.
Here is the JS I have come up with. It returns 3 which is not correct as you can see in the table.

const storage = $("Tire Checks.Rows");

const count = storage.filter(item => item.LF >= '100').length;
 return(count);

![image](https://user-images.githubusercontent.com/94488623/145488009-add4cdf9-1abf-4698-8de2-c5740b77bf2b.png)

**Comment by mjashanks** [comment]
Try
`const count = storage.filter(item => item.LF >= 100).length;`

I've removed the quotations from `100`, as it is a number, not  text.

**Reply by twilson9034** [reply]
I double checked my column and it was a text type, so I created a new column of number type and have noticed it has the correct count; however I was using this within a bar chart so it was only showing the total count of that page. Not sure if that is a bug or not?

**Reply by twilson9034** [reply]
@mjashanks is there a way for this to run through all columns not just the named item and return the count of >= 80?

**Reply by mjashanks** [reply]
yea - depends on the exact need.

If you want to check that **any of the columns** are`>=100` ...

`const count = storage.filter(item => item.Column1 >= 100 || item.Column2 >= 100).length;`

`||` means `or`

**Reply by twilson9034** [reply]
@mjashanks when I use this I am still getting an incorrect count. I have gone in and set everything else to 100 just to verify. It seems that its taking the count of all the rows in the table if I use <= ( less than or equal to ) correct? ...  screen shots below are looking at column RFO which is the first pinkish bar, 100, 100, 100, 80, 110. The number below the bar chart is the return(count). All my columns are text columns. 
![image](https://user-images.githubusercontent.com/94488623/146046471-080dccab-95c9-4177-b0a7-aff7645c2fa8.png)
![image](https://user-images.githubusercontent.com/94488623/146046547-5c06f8d6-f5a7-435c-b2bd-d2d8bc027bd5.png)

Below is if I change it to >= ( greater than or equal to ) correct?...
![image](https://user-images.githubusercontent.com/94488623/146047444-069cc4ff-54e9-46f8-b780-c0f2e6ff9e43.png)
![image](https://user-images.githubusercontent.com/94488623/146046933-e4063daa-fa7a-4b40-9546-326b29beaf03.png)

**Reply by mjashanks** [reply]
of item.RFO is a Text column, then Javascript is doing the calculation `"100" >= "80"` - which returns `false`. It seems strange, but its because both values are text and not numbers

i.e. 
`100 >= 80` returns `true`
 `"100" >= "80"` returns `false`

**Reply by twilson9034** [reply]
ok I will rebuild the table with number columns and see if that will change anything.

**Reply by twilson9034** [reply]
@mjashanks I changed all columns over to numbers and its still not acting the way I think it should.  .length should give me the total amount of within all columns correct? Here is the app if you don't mind taking a look at it and seeing if I have something set wrong.
Data table is "Tire Check", Screen is / > Home, Data provider is Tire check with the Javascript in New paragraph.
The way it is setup right now I should expect to be getting a total count of 50. Correct me if I am wrong once you look at it. 
[Spinnaker Oil Maintenance-export-1639505856613.txt](https://github.com/Budibase/budibase/files/7713888/Spinnaker.Oil.Maintenance-export-1639505856613.txt)

**Reply by twilson9034** [reply]
It seems that it is just returning the count for how many rows have a true in the filter. Not returning the actual count of items within those rows. Is it possible?

**Comment by twilson9034** [comment, marked as answer]
Might be an easier way but this is what I found and came up with and give the Total count of all object less than a number. My apologize if I wasn't clear in what I was looking for.
```
tirechecks = $("Tire Check.Rows")

var LFcount = tirechecks.reduce(function (n, tire) {
    return n + (tire.LF < 80);
}, 0);
var LFOcount = tirechecks.reduce(function (n, tire) {
    return n + (tire.LFO < 80);
}, 0);
var LFIcount = tirechecks.reduce(function (n, tire) {
    return n + (tire.LFI < 80);
}, 0);
var RFOcount = tirechecks.reduce(function (n, tire) {
    return n + (tire.RFO < 80);
}, 0);
var RFIcount = tirechecks.reduce(function (n, tire) {
    return n + (tire.RFI < 80);
}, 0);
var LROcount = tirechecks.reduce(function (n, tire) {
    return n + (tire.LRO < 80);
}, 0);
var LRIcount = tirechecks.reduce(function (n, tire) {
    return n + (tire.LRI < 80);
}, 0);
var RROcount = tirechecks.reduce(function (n, tire) {
    return n + (tire.RRO < 80);
}, 0);
var RRIcount = tirechecks.reduce(function (n, tire) {
    return n + (tire.RRI < 80);
}, 0);
var RFcount = tirechecks.reduce(function (n, tire) {
    return n + (tire.RF < 80);
}, 0);


return(LFcount) + (LFOcount) + (LFIcount) + (RFOcount) + (RFIcount) + (LROcount) + (LRIcount) + (RROcount) + (RRIcount) + (RFcount) + "Tires are less than 80 PSI";
```
